src/nrtk_explorer/app/transforms.py

```diff
@@ -295,7 +295,7 @@ async def _update_images(self):
             self.state.hovered_id = ""
 
         for selected_id in selected_ids:
-            filename = get_image_path(selected_id)
+            filename = get_dataset(self.state.current_dataset).get_image_fpath(selected_id)
```

Comment:
I'm guilty of propigating this too, but it would be nice if the whole app did not really care how we store the `current_dataset`.  This should be an internal detail of the `dataset.py` module.  We may want to change `state.current_dataset` to just be human readable thing rather than file path.

So downstream code could just call `get_dataset().get_image_fpath(id)` or event cleaner just `get_image_fpath`

Reply:
I'm not sure about all of it, but I definitely don't like having `get_dataset(self.state.current_dataset)` in many places. Since it is dependent of a server, it seems that it could be attached to its controller like.

```python
def get_current_dataset():
   return get_dataset(self.state.current_dataset)

def get_image_fpath(selected_id):
   return get_current_dataset().get_image_fpath(selected_id)

server.controller. get_image_fpath = get_image_fpath
```

And then anyone with a server can have access to it

```python
@property
def get_image_fpath():
    return self.server.controller.get_image_fpath
```

Reply:
I went the @jourdain route, find it changed in the last push